ros/api/v0/hosts.py

```diff
@@ -92,8 +92,11 @@ def get(self):
         for row in query_results:
             system_dict = row.System.__dict__
             host = {skey: system_dict[skey] for skey in system_columns}
-            host['recommendation_count'] = len(host['rule_hit_details'])
-            host['state'] = host['rule_hit_details'][0].get('key')
+            if host['rule_hit_details']:
+                host['recommendation_count'] = len(host['rule_hit_details'])
+                host['state'] = host['rule_hit_details'][0].get('key')
```

Comment:
@patilsuraj767, 
[Query] As we have one rule set which contains all rules, each rule hit will be having `key` value.
On frontend, each system row is having `state` column.  
Here you assigned `host['rule_hit_details'][0].get('key')` value to `state`, does that mean every `key` from rule hit contain same value?  OR need a additional logic to get `state`?

Reply:
@kgaikwad  As discussed internally, for now we will be only having one rule hit.